### PR TITLE
refactor(yabloc_pose_initializer): use cpp DNN module instead of python

### PR DESCRIPTION
**Pull request by KYabuuchi**
## Description
* Replaced python scripts which is used for DNN with C++ node.
* Removed the semantic_segmentation service that was only used between python and C++ nodes in yabloc_pose_initializer.

With this change, YabLoc no longer relies on Python at all.

## Tests performed

I verified it with logging_simulator.launch.xml using [this rosbag](https://drive.google.com/file/d/1UqULyfidxcA5JidfHWAsSqNy8itampAX/view?usp=sharing).  The initialization works as before.

image

The colored arrows displayed in RViz is `/localization/pose_estimator/yabloc/initializer/init_candidates`.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: localization/yabloc/yabloc_pose_initializer/CMakeLists.txt
Comment (kminoda) [inline localization/yabloc/yabloc_pose_initializer/CMakeLists.txt:29]: Please remove SemanticSegmentation.srv related documents from Readme

**Review by kminoda — COMMENTED** [review]
LGTM except the following comment :+1:

File: localization/yabloc/yabloc_pose_initializer/CMakeLists.txt
Comment (KYabuuchi) [reply localization/yabloc/yabloc_pose_initializer/CMakeLists.txt:29]: Thanks for your reviewing :heart: I fixed it in [cfe971c](https://github.com/autowarefoundation/autoware.universe/pull/5025/commits/cfe971c9511234c3e418efd4d8536019680e6931)

**Review by kminoda — APPROVED** [review]
LGTM, THANK YOU!!

**Comment by KYabuuchi** [comment]
@Motsu-san Could you approve this PR? 
Minoda-san has already reviewed this PR and I need another code-owner's approval to merge.

**Review by Motsu-san — APPROVED** [review]
Thank you for your courteous work!

**Pull request merged**
